server/app/services/ProgramBlockValidation.java

```diff
@@ -46,28 +53,30 @@ public enum AddQuestionResult {
    * <p>This should be checked both during rendering to make sure admins don't see ineligible
    * questions when creating blocks. Also it should be checked during the actual server-side block
    * creating to ensure that no one can maliciously mess up data by sending specially crafted
-   * request (or in case we mess up client-side checks).
+   * request (or in case we mess up client-side checks). It also ensures that a question tombstoned
+   * in another browser tab, doesn't get accidently added to a program.
    */
-  public static AddQuestionResult canAddQuestion(
+  public AddQuestionResult canAddQuestion(
       ProgramDefinition program, BlockDefinition block, QuestionDefinition question) {
+    if (version.getTombstonedQuestionNames().contains(question.getName())) {
+      return AddQuestionResult.QUESTION_TOMBSTONED;
+    }
```

Comment:
**suggestion (non-blocking):** in this or a follow up PR, it'd also be good to ensure that the question IS in either the current draft version or current active version.

Reply:
I am new to this side of the code. for your comment on "in this or a follow up PR, it'd also be good to ensure that the question IS in either the current draft version or current active version." --> Is this information available in the version object or should I be looking else where? I see this class "ActiveAndDraftQuestions"has a list of all active and draft questions

Reply:
It isn't available on `Version`, so you'll likely need to inject something else and pass it in through the factory, fine to do in a separate PR or I can do it if you'd rather